src/core_types.jl

```diff
@@ -1,14 +1,20 @@
 """
-    abstract Rotation{T} <: StaticMatrix{T}
+    abstract Rotation{N,T} <: StaticMatrix{T}
 
-An abstract type representing 3D rotations. More abstractly, they represent
-3×3 unitary (orthogonal) matrices.
+An abstract type representing `N`-dimensional rotations. More abstractly, they represent
+unitary (orthogonal) `N`×`N` matrices.
 """
-abstract Rotation{T} <: StaticMatrix{T}
+abstract Rotation{N,T} <: StaticMatrix{T}
 
-Base.size{R <: Rotation}(::Union{R,Type{R}}) = (3,3)
+typealias Rotation2{T} Rotation{2,T}
```

Comment:
It's not a typing thing intended for users (it's not exported), it's an internal make-dispatch-easier-until-julia-gets-triangular-dispatch thing. See `rand`. The alternative is
```jl
valdims{N,T}(::Type{Rotation{N,T}}) = Val{N}
rand{R<:Rotation}(::Type{R}) = _rand(R, valdims(R))
function _rand{R<:Rotation}(::Type{R}, ::Type{Val{2}})   ...  end
function _rand{R<:Rotation}(::Type{R}, ::Type{Val{3}})   ...  end
```
which I think is harder to understand.